app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx

```diff
@@ -0,0 +1,256 @@
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { myRemoveCamelCase } from 'services/filters/removeCamelCase';
+import AvailablePluginsStore from 'services/AvailablePluginsStore';
+import { useOnUnmount } from 'services/react/customHooks/useOnUnmount';
+import { shouldShowCustomIcon, getCustomIconSrc, filterPlugins, generateLabel } from './helpers';
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Button,
+  Chip,
+  Icon,
+  ListItem,
+  Typography,
+} from '@material-ui/core';
+import AppsIcon from '@material-ui/icons/Apps';
+import ListIcon from '@material-ui/icons/List';
+import styled, { css } from 'styled-components';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+
+interface ISidePanelProps {
+  itemGenericName: string;
+  groups: any;
+  groupGenericName: string;
+  onPanelItemClick: any;
+}
+
+export const SidePanel = ({
+  itemGenericName,
+  groups,
+  groupGenericName,
+  onPanelItemClick,
+}: ISidePanelProps) => {
+  const [searchText, setSearchText] = useState<string | undefined>(undefined);
+  const [sidePanelViewType, setSidePanelViewType] = useState<string>('icon');
+  const groupWrapperHeight = 'calc(100% - ' + (groups.length * 35 - 35 - 1) + 'px)';
+  const [pluginsMap, setPluginsMap] = useState({});
+  let AvailablePluginsStoreSubscription;
+
+  useEffect(() => {
+    AvailablePluginsStoreSubscription = AvailablePluginsStore.subscribe(() => {
+      const all = AvailablePluginsStore.getState();
+      if (all.plugins) {
+        setPluginsMap(all.plugins.pluginsMap);
+      }
+    });
+  }, []);
+
+  useOnUnmount(() => {
+    AvailablePluginsStoreSubscription();
+  });
+
+  const handleSetSearch = (text) => {
+    setSearchText(text);
+  };
+
+  const handleSetSidePanelViewType = (viewType: string) => {
+    setSidePanelViewType(viewType);
+  };
+
+  const handleClickPlugin = (event, plugin) => {
+    // the angular controller function has event as an input
+    onPanelItemClick(event, plugin);
+  };
+
+  const renderPlugins = (plugins: [any]) => {
+    return plugins.map((plugin) => {
+      const showCustomIcon = shouldShowCustomIcon(plugin, pluginsMap);
+      return (
+        <div
+          role="button"
+          className={`plugin-item ${plugin.nodeClass ? plugin.nodeClass : ''} ${
+            plugin.hovering ? 'hovered' : ''
+          }`}
+          // need to implement my popover
+          // my-popover
```

Comment:
Did you get the popover working for this? Is this comment still needed?

Reply:
@njbriggs popover is being tracked in another ticket